### ISSUE TITLE
pydna function primer_design() breaks calling SeqRecord reverse_complement

### ISSUE THREAD

**Issue by nick-chavez23** [biopython/biopython#4177]
### Setup
I am reporting a problem with Biopython version, Python version, and operating
system as follows:

3.7.15 (default, Oct 12 2022, 19:14:55)
[GCC 7.5.0]
CPython
Linux-5.10.133+-x86_64-with-Ubuntu-18.04-bionic
1.80

### Expected behaviour
function SeqRecord() creates a SeqRecord object 

### Actual behaviour
```python
[/usr/local/lib/python3.7/dist-packages/Bio/SeqRecord.py](https://localhost:8080/#) in reverse_complement(self, id, name, description, features, annotations, letter_annotations, dbxrefs)
1234 # Default to DNA)
1235 seq = self.seq.reverse_complement(
-> 1236 inplace=False
1237 ) # TODO: remove inplace=False
1238 if isinstance(self.seq, MutableSeq):


TypeError: reverse_complement() got an unexpected keyword argument 'inplace'
```

**Comment by mdehoon** [comment]
What is your code?

**Comment by mdehoon** [comment]
@peterjc 
`MutableSeq` does have the `inplace` argument:
```python
>>> from Bio.Seq import MutableSeq
>>> m = MutableSeq("AACC")
>>> m.reverse_complement(inplace=False)
MutableSeq('GGTT')
>>> m
MutableSeq('AACC')
>>> m.reverse_complement(inplace=True)
MutableSeq('GGTT')
>>> m
MutableSeq('GGTT')
```

**Comment by peterjc** [comment]
Sorry, must have had the wrong version of the file opened or something.

So we still don't know how this error is being triggered.

**Comment by nick-chavez23** [comment]
I have a script that uses both pydna and Biopython. pydna has a function called primer_design() that when run, uses SeqRecord() and raises this error. 

When I use version 1.79 of Biopython, the issue does no occur. It looks like the region of code that the error comes from is from your original comment about being a MutableSeq object. I am not sure what the issue is but something does seem to have changed in the release of Biopython 1.80.

**Comment by peterjc** [comment]
Cross reference https://github.com/BjornFJohansson/pydna/issues/90

Which version(s) of pydna are failing?

**Comment by nick-chavez23** [comment]
I have been using version 4.0.7 of pydna for the project I am working on.

**Comment by mdehoon** [comment]
@nick-chavez23  Can you show a complete but minimal example of your code triggering the error?

**Comment by nick-chavez23** [comment]
I have written a very simple function that takes in a sequence and outputs an optimized primer pair based on our desired melting temperature and length of primer. The code is as follow:

```
def FindPrimer(Sequence, rev = None):
  Primers = primer_design(Dseqrecord(Sequence), limit = 18, target_tm = 60, rp = rev)
  fPrim   = Primers.forward_primer.seq
  rPrim   = Primers.reverse_primer.seq
  return [fPrim, rPrim]
```

**Comment by nick-chavez23** [comment]
I have the following imports when I run my script:

```
!pip install biopython
!pip install pydna

from Bio import SeqIO
from Bio.Seq import Seq
from Bio.SeqRecord import SeqRecord
from Bio.SeqFeature import SeqFeature, FeatureLocation
from pydna.dseqrecord import Dseqrecord
from pydna.design import primer_design
```

The input Sequence to my function FindPrimer() is a string type object.

**Comment by peterjc** [comment]
Thank you. Now we have a minimal self contained example which fails:

```
from pydna.dseqrecord import Dseqrecord
from pydna.design import primer_design

primer_design(Dseqrecord("ACGT" * 100), limit=18, target_tm=60, rp=None)
```

Its something pydna need to fix I think, Biopython 1.80 changed the ``Seq`` (etc) object's ``.reverse_complement(...)`` signature to:

```python
    def reverse_complement(self, inplace=None):
```

i.e. In their subclass, pydna need to add the new argument to match.
